aws/endpoints.go

```diff
@@ -54,8 +54,27 @@ type Endpoint struct {
 	// The signing method that should be used for signing the requests to the
 	// endpoint.
 	SigningMethod string
+
+	// The source of the Endpoint. By default, this will be ServiceMetadata.
+	// When providing a custom endpoint, you should set the source as CustomSource.
+	// If source is not provided when providing a custom endpoint, the SDK may not
+	// perform required host mutations correctly. EndpointSource should be used along with
+	// HostnameImmutable property as per the usage requirement.
+	EndpointSource EndpointSource
```

Comment:
`EndpointSource` as the parameter name is a bit redundant since the type is `Endpoint`. e.g. `Endpoint.EndpointSource`  Could name this member just `Source`.